### PR TITLE
[CLOUDTRUST-2297] Add flag for user accreditations

### PR DESCRIPTION
**Pull request by fperot74**
User accreditations are updated in /kyc and /validation
Accreditations are sent through the "GetUser" API of /kyc, /management and /users

### CODE REVIEW

File: internal/keycloakb/accreditationsmodule.go [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+	err = am.keycloakClient.UpdateUser(accessToken, realmName, userID, kcUser)
+	if err != nil {
+		am.logger.Warn(ctx, "msg", "CreateAccreditations: can't update Keycloak user", "err", err.Error(), "realm", realmName, "user", userID)
+		return err
+	}
+
+	return nil
+}
+
+func arrayContains(slice []string, value string) bool {
```
Comment (bsoniam) [inline internal/keycloakb/accreditationsmodule.go]: used again here

Comment (fperot74) [reply internal/keycloakb/accreditationsmodule.go]: I removed this function as validation.IsStringInSlice(...) is now merged in common-service

File: internal/keycloakb/datehelpers.go [hunk 2]
```diff
@@ -46,3 +47,87 @@ func NextMonth(ref time.Time) time.Time {
 	}
 	return time.Date(year, month, 1, 0, 0, 0, 0, ref.Location()).UTC()
 }
+
+type longDuration struct {
```
Comment (harture) [inline internal/keycloakb/datehelpers.go]: All those changes are part of another PR in common-service

File: api/management/api.go [hunk 3]
```diff
@@ -525,9 +525,9 @@ func (config RealmCustomConfiguration) Validate() error {
 }
 
 // Validate is a validator for RealmAdminConfiguration
-func (config RealmAdminConfiguration) Validate() error {
+func (rac *RealmAdminConfiguration) Validate() error {
```
Comment (harture) [inline api/management/api.go]: Pointeur ?

File: internal/keycloakb/datehelpers_test.go [hunk 4]
```diff
@@ -106,3 +106,40 @@ func TestNextMonth(t *testing.T) {
 	assert.Equal(t, nextMonthIndia, NextMonth(reference.In(locIndia)))
 	assert.Equal(t, nextMonthCookIsland, NextMonth(reference.In(locCookIsland)))
 }
+
+func TestLongDuration(t *testing.T) {
```
Comment (harture) [inline internal/keycloakb/datehelpers_test.go]: those helpers have been moved in common

File: pkg/validation/component.go [hunk 5]
```diff
@@ -212,9 +230,24 @@ func needDBUserUpdate(user api.UserRepresentation) bool {
 }
 
 func (c *component) CreateCheck(ctx context.Context, userID string, check api.CheckRepresentation) error {
+	var accessToken string
+	var err error
+
+	if check.IsIdentificationSuccessful() {
```
Comment (harture) [inline pkg/validation/component.go]: I think we should prevent a scenario where we have an accreditation without persisting the check. Maybe we should swap between persistency of accrediation and check ? WHat do you think about it ?

File: internal/keycloakb/accreditationsmodule.go [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+}
+
+func (am *accredsModule) CreateAccreditations(ctx context.Context, accessToken, realmName, userID, condition string) error {
+	var kcUser, count, err = am.GetUserAndPrepareAccreditations(ctx, accessToken, realmName, userID, condition)
+	if err != nil || count == 0 {
+		return err
+	}
+
+	// Update user in Keycloak
+	err = am.keycloakClient.UpdateUser(accessToken, realmName, userID, kcUser)
```
Comment (harture) [inline internal/keycloakb/accreditationsmodule.go]: Is it really the matter of this module to update the user ?
Maybe this accreditations component could only be responsible to create the array of accreditations to persists into the user and let the upper layer (component) to update the user

**Pull request merged**